docker-for-mac/mutagen-caching.md

```diff
@@ -0,0 +1,111 @@
+---
+description: High-performance two-way file sync for volume mounts
+keywords: mac, mutagen, volumes, file sync,
+title: Mutagen-based caching
+---
+
+Docker Desktop for Mac on Edge has a new file sharing feature which performs
+a continuous two-way sync of files between the host and containers using
+[Mutagen](https://mutagen.io/). This feature is ideal for app development where:
+
+- the source code tree is quite large
+- the source is edited on Mac
+- the source is compiled and run interactively inside Linux containers.
+
+The following section contains an example to show how the Mutagen feature should be used to sync files between the host and containers .
+
+## A simple React app
+
+The following example bootstraps a simple React app with `npx` and configures
+Docker Desktop to sync the source code between the host and a
+development container.
+
+First, create a directory which will contain the app:
+
+```
+$ mkdir ~/workspace/my-app
+```
+Next, enable the two-way file sync feature in Docker Desktop:
+
+1. From the Docker Desktop menu, click **Preferences** > **Resources** > **File sharing**.
+
+2. Type the new directory name in the bottom of the list and press the enter key.
+
+3. Click on the toggle next to the directory name.
+
+4.  Click **Apply & Restart** for the changes to take effect.
+
+When Docker Desktop has restarted, the **File sharing** page looks like
+this:
+
+![Caching with mutagen is "Ready"](images/mac-mutagen-ready.png)
+
+Run the following command to start a container and bootstrap the app with `npx`:
+
+```
+$ docker run -it -v ~/workspace/my-app:/my-app -w /my-app -p 3000:3000 node:lts bash
+root@95441305251a:/my-app# npx create-react-app app
+root@95441305251a:/my-app# cd app
+root@95441305251a:/my-app# npm start
+```
+
+Once the development webserver has started, open [https://localhost:3000/](https://localhost:3000/) in
+your browser and observe the app is running.
+
+Return to the **File sharing** page in the UI and observe the status of the cache toggle located next to the directory name. The status will be
+updated as file changes are detected and then synchronized between the host
+and the containers.
+
+Wait until the text says **Ready** and then open the source code in your IDE on
+the host. Edit the file `src/App.js`, save the changes and observe the change
+on the webserver.
+
+As you edit code on the host, the changes are detected and transferred to the
+container for testing. Changes inside the container (for example, the creation of build artifacts) are detected and transferred back to the host.
+
+## Avoiding synchronizing a subdirectory
```

Comment:
It feels like this section should be a child of Best Practices. But maybe it can't work because it's too long?

Reply:
Yea, I'd leave it as a separate section as it also calls out an important use case?